atlasdb-cassandra/src/main/java/com/palantir/atlasdb/cassandra/CassandraAtlasDbFactory.java

```diff
@@ -84,7 +84,10 @@ static CassandraKeyValueServiceConfig preprocessKvsConfig(
         CassandraKeyValueServiceConfig configWithNamespace =
                 CassandraKeyValueServiceConfigs.copyWithKeyspace(cassandraConfig, desiredKeyspace);
 
-        return new CassandraReloadableKvsConfig(configWithNamespace, runtimeConfig);
+        CassandraReloadableKvsConfig reloadableKvsConfig =
+                new CassandraReloadableKvsConfig(configWithNamespace, runtimeConfig);
+        reloadableKvsConfig.check();
```

Comment:
This method should not be called in callers - it should not be their respsonsibility. If you want to keep it, then it should be called in the constructor.

However I'd strongly prefer to just remove it. It's doesn't make sense to have a precondition method on this non-immutable class since updates to the underlying values won't be checked (unlike runtime configuration classes which are instantiated every time new config is loaded).

If we want to have a check method, it belongs on the `CassandraKeyValueServiceRuntimeConfig` class.

Reply:
We can't really check Runtime either though right? Because people could still have it on Install.

I think when I suggested moving the check what I had in mind was doing it in the constructor, though you have a good point that updates are not checked.

Fine with removing the checks at creation time. We might want `servers()` to throw if it's going to return an empty, as I don't imagine users would expect that case.

Reply:
Are we not worried about runtime errors in the places that call `servers()`?

Reply:
Maybe it's worth adding an `InvalidConfig` variant to `CassandraServersConfig` to force callers to handle this case.

Most uses are `ThriftHostsExtractingVisitor`. We can either:
- Change the return type to `Optional<Set<InetSocketAddress>>` and let callers handle an empty optional
- Just throw in `ThriftHostsExtractingVisitor`